packages/login/network/BitcoinMiddlewareNode.js

```diff
@@ -84,10 +85,43 @@ export default class BitcoinMiddlewareNode extends BitcoinAbstractNode {
     }
   }
 
-  async getTransactionsList (/*address*/) {
+  async getTransactionsList (address, id, skip, offset) {
+
+    try {
+      const test = await this._api.get(`tx/${address}/history?skip=0&limit=1`)
+      if (test.status === 200) {
+        return this._getTransferFromMiddleware(address, skip, offset)
+      }
+    } catch (e) {
+      // eslint-disable-next-line
+      console.warn('Middleware API is not available, fallback to block-by-block scanning', e)
+    }
+
     return []
   }
 
+  async _getTransferFromMiddleware (account: string, skip: number, offset: number): Array<TxModel> {
+    let txs = []
+    const url = `tx/${account}/history?skip=${skip}&limit=${offset}`
+    try {
+      const result = await this._api.get(url)
+      if (typeof result !== 'object' || !result.data) {
+        throw new Error('invalid result')
+      }
+      if (result.status !== 200) {
```

Comment:
Status can be 304, remove status check.
Axios know better how to interpret status code